common-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java

```diff
@@ -70,120 +70,96 @@ public static String modifyMultipleMatchKeyword(String query){
     public static final String STANDARD_CONCEPT_CODE = "S";
     public static final String CLASSIFICATION_CONCEPT_CODE = "C";
 
-    public Slice<Concept> searchConcepts(String query,
-                                         StandardConceptFilter standardConceptFilter, List<String> vocabularyIds,
-                                         List<String> domainIds, int limit, int minCount) {
-
+    public Slice<Concept> searchConcepts(String query, StandardConceptFilter standardConceptFilter, List<String> vocabularyIds, List<String> domainIds, int limit, int minCount, List<Long> synonymConceptIds) {
 
         Specification<Concept> conceptSpecification =
                 (root, criteriaQuery, criteriaBuilder) -> {
                     List<Predicate> predicates = new ArrayList<>();
-
-                    // Check that the concept name, code, or ID matches the query string.
-                    List<Predicate> conceptCodeIDName = new ArrayList<>();
-                    Expression<Double> matchExp = null;
-
-                    if(!Strings.isNullOrEmpty(query)){
-
-                        final String keyword = modifyMultipleMatchKeyword(query);
-
-                        conceptCodeIDName.add(criteriaBuilder.equal(root.get("conceptCode"),
-                                criteriaBuilder.literal(query)));
-
-                        try {
-                            long conceptId = Long.parseLong(query);
-                            conceptCodeIDName.add(criteriaBuilder.equal(root.get("conceptId"),
-                                    criteriaBuilder.literal(conceptId)));
-                        } catch (NumberFormatException e) {
-                            // Not a long, don't try to match it to a concept ID.
+                    List<Predicate> standardConceptPredicates = new ArrayList<>();
+                    standardConceptPredicates.add(criteriaBuilder.equal(root.get("standardConcept"),
+                            criteriaBuilder.literal(STANDARD_CONCEPT_CODE)));
+                    standardConceptPredicates.add(criteriaBuilder.equal(root.get("standardConcept"),
+                            criteriaBuilder.literal(CLASSIFICATION_CONCEPT_CODE)));
+
+                    List<Predicate> nonStandardConceptPredicates = new ArrayList<>();
+                    nonStandardConceptPredicates.add(criteriaBuilder.notEqual(root.get("standardConcept"),
+                            criteriaBuilder.literal(STANDARD_CONCEPT_CODE)));
+                    nonStandardConceptPredicates.add(criteriaBuilder.notEqual(root.get("standardConcept"),
+                            criteriaBuilder.literal(CLASSIFICATION_CONCEPT_CODE)));
+
+                    // If the concepts that match the synonym names are found, they are fetched by applying standard concept filer, domain and vocabulary id filters
```

Comment:
Nit: Comment is a bit misleading. The synonms don't have anything to do with vocabulary and other filters. I would just say : Apply the synomns filter if we have them or apply query/concept_code/id filters. They are mutually exclusive.  

Also, the standard concept filter could be applied generically it looks like  as you do the vocabs and domain ? Do you think ?

Reply:
cannot move the standard concept filter as for concept name and code/id match match had to loop it together along with the match.